From 8612d7a7c87f8de337ddcc6e08327c34ffb748f9 Mon Sep 17 00:00:00 2001
From: Philip Balister <philip@balister.org>
Date: Tue, 23 Sep 2014 13:18:41 -0400
Subject: [PATCH] Disable pulseaudio.

Signed-off-by: Philip Balister <philip@balister.org>
---
 gqrx.pro | 8 --------
 1 file changed, 8 deletions(-)

diff --git a/gqrx.pro b/gqrx.pro
index 1d9520d..31e8b01 100644
--- a/gqrx.pro
+++ b/gqrx.pro
@@ -27,14 +27,6 @@ macx {
 # enable pkg-config to find dependencies
 CONFIG += link_pkgconfig
 
-unix:!macx {
-    packagesExist(libpulse libpulse-simple) {
-        # Comment out to use gr-audio (not recommended with ALSA and Funcube Dongle Pro)
-        AUDIO_BACKEND = pulse
-        message("Gqrx configured with pulseaudio backend")
-    }
-}
-
 RESOURCES += \
     resources/icons.qrc \
     resources/textfiles.qrc
-- 
2.7.4

