From d230397957603a026c688dd462d349f8dcb645c8 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ji=C5=99=C3=AD=20Pinkava?= <j-pi@seznam.cz>
Date: Mon, 15 Feb 2016 14:22:04 +0100
Subject: [PATCH] allow both 3.7.* and git version of GnuRadio

---
 CMakeLists.txt | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index f0b1283..212a30e 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -150,7 +150,12 @@ set(GRC_BLOCKS_DIR      ${GR_PKG_DATA_DIR}/grc/blocks)
 # Find build dependencies
 ########################################################################
 set(GR_REQUIRED_COMPONENTS RUNTIME PMT BLOCKS)
-find_package(Gnuradio 3.7.3 REQUIRED)
+set(MIN_GR_VERSION "3.7.3")
+find_package(Gnuradio REQUIRED)
+if("${Gnuradio_VERSION}" VERSION_LESS MIN_GR_VERSION)
+    MESSAGE(FATAL_ERROR "GnuRadio version required: >=\"" ${MIN_GR_VERSION} "\" found: \"" ${Gnuradio_VERSION} "\"")
+endif()
+
 find_package(GnuradioIQBalance)
 find_package(UHD)
 find_package(GnuradioUHD)
-- 
2.7.4

