From f2ed093da93811745b432ca8eb3f6434bc29aab7 Mon Sep 17 00:00:00 2001
From: Andrej Rode <mail@andrejro.de>
Date: Wed, 17 Oct 2018 02:33:00 +0200
Subject: [PATCH 2/2] fix

---
 gnuradio-runtime/lib/block.cc | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/gnuradio-runtime/lib/block.cc b/gnuradio-runtime/lib/block.cc
index c86e79212..b6d926c44 100644
--- a/gnuradio-runtime/lib/block.cc
+++ b/gnuradio-runtime/lib/block.cc
@@ -194,13 +194,16 @@ namespace gr {
   void
   block::set_relative_rate(uint64_t interpolation, uint64_t decimation)
   {
+    mpz_class interp, decim;
     if (interpolation < 1)
       throw std::invalid_argument("block::set_relative_rate: interpolation rate cannot be 0");
 
     if (decimation < 1)
       throw std::invalid_argument("block::set_relative_rate: decimation rate cannot be 0");
 
-    d_mp_relative_rate = mpq_class(interpolation, decimation);
+    mpz_import(interp.get_mpz_t(), 1, 1, sizeof(interpolation), 0, 0, &interpolation);
+    mpz_import(decim.get_mpz_t(), 1, 1, sizeof(decimation), 0, 0, &decimation);
+    d_mp_relative_rate = interp / decim;
     d_mp_relative_rate.canonicalize();
     d_relative_rate = d_mp_relative_rate.get_d();
   }
-- 
2.14.4

