From 8bb63e7ea51eefbc9ada03b6e71ee9425a931534 Mon Sep 17 00:00:00 2001
From: Philip Balister <philip@balister.org>
Date: Wed, 6 Sep 2023 10:42:23 -0700
Subject: [PATCH] Fix sse2neon.h build on gcc-13.

Signed-off-by: Philip Balister <philip@balister.org>
---
 include/volk/sse2neon.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/include/volk/sse2neon.h b/include/volk/sse2neon.h
index c7baeab..6ce784b 100644
--- a/include/volk/sse2neon.h
+++ b/include/volk/sse2neon.h
@@ -575,7 +575,7 @@ FORCE_INLINE uint32_t _mm_crc32_u8(uint32_t, uint8_t);
 
 // Older gcc does not define vld1q_u8_x4 type
 #if defined(__GNUC__) && !defined(__clang__) &&                        \
-    ((__GNUC__ <= 12 && defined(__arm__)) ||                           \
+    ((__GNUC__ <= 13 && defined(__arm__)) ||                           \
      (__GNUC__ == 10 && __GNUC_MINOR__ < 3 && defined(__aarch64__)) || \
      (__GNUC__ <= 9 && defined(__aarch64__)))
 FORCE_INLINE uint8x16x4_t _sse2neon_vld1q_u8_x4(const uint8_t *p)
-- 
2.41.0

