From 614b21e53145aea739627c08f2b04d87cae5bb17 Mon Sep 17 00:00:00 2001
From: Philip Balister <philip@balister.org>
Date: Fri, 23 Sep 2022 17:20:04 -0400
Subject: [PATCH] Fix build for 32 bit arm with neon.

Signed-off-by: Philip Balister <philip@balister.org>
---
 include/volk/sse2neon.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/include/volk/sse2neon.h b/include/volk/sse2neon.h
index ff4857f..29db305 100644
--- a/include/volk/sse2neon.h
+++ b/include/volk/sse2neon.h
@@ -444,7 +444,7 @@ FORCE_INLINE uint32_t _mm_crc32_u8(uint32_t, uint8_t);
 
 // Older gcc does not define vld1q_u8_x4 type
 #if defined(__GNUC__) && !defined(__clang__) &&                        \
-    ((__GNUC__ <= 11 && defined(__arm__)) ||                           \
+    ((__GNUC__ <= 12 && defined(__arm__)) ||                           \
      (__GNUC__ == 10 && __GNUC_MINOR__ < 3 && defined(__aarch64__)) || \
      (__GNUC__ <= 9 && defined(__aarch64__)))
 FORCE_INLINE uint8x16x4_t _sse2neon_vld1q_u8_x4(const uint8_t *p)
-- 
2.37.3

